tests/unit/test_charm.py

```diff
@@ -83,11 +85,50 @@ def test_given_private_key_is_stored_when_certificates_relation_joined_then_cert
         self.harness.add_relation_unit(
             relation_id=relation_id, remote_unit_name="certificates-provider/0"
         )
-
+        unit_number = self.harness.model.unit.name.split("/")[1]
+        patch_generate_csr.assert_called_with(
+            private_key=PRIVATE_KEY.encode(),
+            private_key_password=PRIVATE_KEY_PASSWORD.encode(),
+            subject=f"{self.harness.charm.app.name}-{unit_number}.{self.harness.model.name}",
+        )
         patch_request_certificate_creation.assert_called_with(
             certificate_signing_request=CSR.encode()
         )
 
+    @patch("charm.generate_csr")
+    @patch(
+        "charms.tls_certificates_interface.v2.tls_certificates.TLSCertificatesRequiresV2.request_certificate_creation"  # noqa: E501, W505
+    )
+    def test_given_common_name_config_is_set_when_certificates_relation_joined_then_certificate_is_requested(  # noqa: E501
+        self,
+        _,
+        patch_generate_csr,
```

Comment:
We are patching `request_certificate_creation` and the name of the test suggests asserting for requesting a certificate, however the name of the variable is `patch_generate_csr`, I think it's a bit confusing.